File: Demos/Visual Basic .Net/09. Addendas/Adenda/Corporate Travel Services.vb
Imports HyperSoft.Ejemplo.Data
Imports HyperSoft.ElectronicDocumentLibrary.CorporateTravelServices.Addenda

Friend NotInheritable Partial Class Adenda
	
	Friend Shared Function CorporateTravelServices(ByRef fileName As String) As Boolean
		'En este método se cargan los datos de la factura.
		Cfdi40.CargarDatosCompleto(electronicDocument)

		Dim addenda As CorporateTravelServices = HyperSoft.ElectronicDocumentLibrary.CorporateTravelServices.Addenda.CorporateTravelServices.NewEntity()

		' Se asignan los datos del complemento
		addenda.Data.Cupon.Value = "123"

		electronicDocument.Data.Addendas.Add(addenda)

		Return Save("Addenda_CorporateTravelServices.xml", fileName)
	End Function

End Class

File: Demos/CSharp/09. Addendas/Adenda/Corporate Travel Services.cs
using HyperSoft.Ejemplo.Data;
using HyperSoft.ElectronicDocumentLibrary.CorporateTravelServices.Addenda;

namespace HyperSoft.Ejemplo.Adendas
{
  internal static partial class Adenda
  {
    internal static bool CorporateTravelServices(out string fileName)
    {
      //En este método se cargan los datos de la factura.
      Cfdi40.CargarDatosCompleto(electronicDocument);

      CorporateTravelServices addenda = HyperSoft.ElectronicDocumentLibrary.CorporateTravelServices.Addenda.CorporateTravelServices.NewEntity();

      // Se asignan los datos del complemento
      addenda.Data.Cupon.Value = "123";

      electronicDocument.Data.Addendas.Add(addenda);

      return Save("Addenda_CorporateTravelServices.xml", out fileName);
    }
  }
}